### PR TITLE
fix: match/resync robustness — name collision, parentId cycle, OPT quoted numerics (#954)

### PR DESCRIPTION
**Pull request by hyiger**
Part of the #954 triage (PR-F: match / resync robustness). Three confirmed P2 correctness gaps, each independent and low-risk.

### 1. Name-tier case-only collision auto-matches an arbitrary filament — `src/lib/matchFilament.ts`
The `name` partial-unique index is case-**sensitive** (no collation), so `PLA Black` and `pla black` can both be active. The name tier resolved with a bare case-insensitive `findOne` and returned whichever Mongo yielded first as a **confident** match — and the SSE scan bus auto-selects a PrusaSlicer preset off `match.name`, so a possibly-wrong preset was picked silently. Now mirrors the `instanceId` tier: **exact case wins**, then a CI fallback that only auto-matches on a single hit; a case-only collision surfaces both as `candidates`. (Same class as the GH #896 vendor-substring fix.)

### 2. Unbounded `parentId` recursion hangs OPT check/sync — `src/lib/resolveEffectiveFilament.ts`
The parent recursion had no seen/depth guard, so a `parentId` cycle in corrupt data (self- or mutual-reference — the API blocks nested inheritance, but a direct DB write / bad migration could create one) recursed forever awaiting `Filament.findOne` and the request never returned. Threads a visited-id set; on revisit it stops and falls back to the raw doc (same degradation as an unloadable parent).

### 3. OPT quoted-numeric YAML → permanent unapplyable resync conflict — `src/lib/openprinttagBrowser.ts`
A **quoted** upstream scalar (`density: "1.24"`) parses to a JS string; the `as number` cast is a compile-time no-op, so the string rode into the Mixed `openprinttagSnapshot` while Mongoose cast the stored field to a number. Every re-sync `check` then compared `1.24` vs `"1.24"` (strict `===`) → a **permanent conflict** that re-surfaces after every apply. Coerce to a real number at the parse boundary (new `toOptNumber`), applied uniformly to all 11 numeric fields; blank/whitespace/garbage → `null` (never `0`, since `Number("")===0`). Legacy string snapshots self-heal on the next sync — no migration. (Same "cast at the boundary" intent as #632 / #894.)

### Tests
Added for each: name case-collision + exact-case-preference (matchFilament), two- and self-node `parentId` cycles that terminate (resolveEffectiveFilament), quoted-numeric coercion + `toOptNumber` edge cases (openprinttagBrowser). Full **coverage** suite green — 99.0% stmt / 97.6% branch, all thresholds held. Typecheck + lint clean.

Part of #954.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by hyiger** [comment]
@codex review

**Comment by hyiger** [comment]
@codex review

File: src/lib/openprinttagBrowser.ts [hunk 1]
```diff
@@ -230,6 +230,27 @@ export function rgbaToHex(rgba: string | undefined | null): string | null {
...
+ * and surfaced a permanent, unapplyable conflict. Coercing here keeps types
+ * honest end-to-end (snapshot, exports, completeness scoring). Empty/whitespace
+ * strings and non-numerics return null — NOT 0 (`Number("") === 0` would
+ * silently invent a value). Same "cast at the boundary" intent as the #632 /
+ * #894 fixes; the OPTMaterial interface already types every field `number | null`.
+ */
+export function toOptNumber(v: unknown): number | null {
+  if (v == null) return null;
+  if (typeof v === "string" && v.trim() === "") return null;
+  const n = Number(v);
```
[the remark this answers is not in the document]
Comment (hyiger) [reply src/lib/openprinttagBrowser.ts]: Fixed in the latest push. `toOptNumber` now coerces only `typeof number` and numeric strings; booleans, arrays, and objects return `null` (absent), so a malformed upstream scalar is skipped by the optResync guard instead of fabricating 0/1. Added test cases for `true`/`false`/`[]`/`[65]`/`{}` → null.

**Comment by hyiger** [comment]
@codex review
